stack/core/src/main/java/org/apache/usergrid/corepersistence/CpRelationManager.java

```diff
@@ -697,25 +695,35 @@ public ConnectionRef createConnection( String connectionType, EntityRef connecte
         final GraphManager gm = managerCache.getGraphManager( applicationScope );
 
 
-        //check if the edge exists
+        //write new edge
 
+        gm.writeEdge( edge ).subscribe();
 
-        final SearchByEdge searchByEdge = new SimpleSearchByEdge(edge.getSourceNode(), edge.getType(), edge.getTargetNode(), Long.MAX_VALUE, SearchByEdgeType.Order.DESCENDING, Optional.absent()  );
+        indexService.queueNewEdge( applicationScope, targetEntity, edge );
 
 
-        //only take 1 and count it.  If we don't have anything, create the edge
-        final int count = gm.loadEdgeVersions( searchByEdge ).take( 1 ).count().toBlocking().last();
+        //now read all older versions of an edge, and remove them.  Finally calling delete
+        final SearchByEdge searchByEdge =
```

Comment:
What are the performance implications of doing this delete every time we write a connection? Seems like every connection write be 2x the time since delete is a heavy operation. Or is there something I'm missing?